xrpl/models/transactions/amm_deposit.py

```diff
@@ -0,0 +1,70 @@
+"""Model for AMMDeposit transaction type."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Dict, Optional
+
+from xrpl.models.amounts import Amount
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import require_kwargs_on_init
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class AMMDeposit(Transaction):
+    """
+    AMDeposit is the deposit transaction used to add liquidity to the AMM instance pool,
+    thus obtaining some share of the instance's pools in the form of LPTokens.
+
+    The following are the recommended valid combinations:
+    - LPTokens
+    - Asset1In
+    - Asset1In and Asset2In
+    - Asset1In and LPTokens
+    - Asset1In and EPrice
+    """
+
+    amm_hash: str = REQUIRED  # type: ignore
+    """
+    AMMHash is a hash that uniquely identifies the AMM instance.
+    """
+
+    lptokens: Optional[Amount] = None
```

Comment:
IMO this should be `lp_tokens`

Reply:
I named it `lptokens` to be consistent with `nftoken` casing but I'm happy to change it to `lp_tokens`